src/installer.ts

```diff
@@ -23,31 +26,26 @@ export async function getJava(
     core.debug(`Tool found in cache ${toolPath}`);
   } else {
     let compressedFileExtension = '';
+
     if (!jdkFile) {
-      core.debug('Downloading JDK from Azul');
-      const http = new httpm.HttpClient('setup-java', undefined, {
-        allowRetries: true,
-        maxRetries: 3
-      });
-      const url = 'https://static.azul.com/zulu/bin/';
-      const response = await http.get(url);
-      const statusCode = response.message.statusCode || 0;
-      if (statusCode < 200 || statusCode > 299) {
-        let body = '';
-        try {
-          body = await response.readBody();
-        } catch (err) {
-          core.debug(`Unable to read body: ${err.message}`);
+      async function setupVendor(distro: string) {
+        switch (distro) {
+          case 'adoptopenjdk':
+            [jdkFile, version] = await getJavaAdoptOpenJDK(
+              version,
+              javaPackage,
+              arch
+            );
+            break;
+          case 'zulu':
+            [jdkFile, version] = await getJavaZulu(version, javaPackage);
+            break;
```

Comment:
I suggest to move this function outside of `getJava` function to improve readability

Reply:
I'd also echo the "change `vendor` to `distro`" and ask to use that everywhere for consistency,  including in the code. So this should be setupDistro()...